app/src/main/contrib/components/Responsibilities/BurdenEstimates/UploadBurdenEstimatesForm.tsx

```diff
@@ -1,86 +1,345 @@
+                       toggle={() => this.resetUploadState(false)}>
+                    {this.state.uploadErrors.length > 0 && this.state.uploadErrors[0].message}
+                </Alert>
+                <Alert color="danger" isOpen={this.props.populateErrors.length > 0}
+                       toggle={this.props.resetPopulateState}>
+                    <p className="render-whitespace">
+                        {this.props.populateErrors.length > 0 && this.props.populateErrors[0].message}
+                    </p>
+                    Please correct the data and re-upload.
+                </Alert>
+                <Alert color="success" isOpen={this.props.hasPopulateSuccess} toggle={this.props.resetPopulateState}>
+                    Success! You have uploaded a new set of burden estimates
+                </Alert>
+                <button disabled={this.submitDisabled()}
+                        id="submit-form"
+                        className="submit start"
+                        onClick={this.startUpload}>Upload
+                </button>
+                <button disabled={!this.state.isUploading} className="cancel"
+                        onClick={this.cancelUpload}>Cancel
+                </button>
+                <small className="form-text text-muted">
+                    * All fields are required
+                </small>
             </div>
+        );
+    }
+}
+
+export const SelectedFile: React.FunctionComponent<UploadEstimatesState> =
+    (props: { file: ResumableFile, fileValidationResult: CustomValidationResult }) => {
+
+        if (props.file) {
+            return <span>
+                <span>File selected: {props.file.fileName}</span>
+                <Alert color="danger" isOpen={!props.fileValidationResult.isValid}
+                       className="mt-3 pathProblems">
+                    {props.fileValidationResult.content}
+                </Alert>
+            </span>;
         } else {
             return null;
         }
-    }
-}
 
-export const mapStateToProps: (state: ContribAppState, props: UploadBurdenEstimatesFormPublicProps)
-    => Partial<UploadBurdenEstimatesFormComponentProps>
-    = (state: ContribAppState, props: UploadBurdenEstimatesFormPublicProps) => {
+    };
+
+export const mapStateToProps: (state: ContribAppState, props: UploadEstimatesPublicProps) => Partial<UploadEstimatesProps>
+    = (state: ContribAppState, props: UploadEstimatesPublicProps) => {
     return {
         ...props,
-        errors: state.estimates.populateErrors,
-        hasSuccess: state.estimates.hasPopulateSuccess
+        url: state.estimates.uploadToken == null || state.estimates.populatingSetId == null ? null :
+            `${settings.apiUrl()}/modelling-groups/${props.groupId}/responsibilities/${props.touchstoneId}/${props.scenarioId}/estimate-sets/${state.estimates.populatingSetId}/actions/upload/${state.estimates.uploadToken}/`,
```

Comment:
The setId is now coming from store state (see https://github.com/vimc/montagu-webapps/pull/399/files#diff-fdc00e5b43ae62249845b96dcc1d4b33R86)